src/tcop/tcop.cpp

```diff
@@ -172,61 +189,77 @@ std::vector<FieldInfoType> TrafficCop::GenerateTupleDescriptor(
 
               if (columns[i].column_type == VALUE_TYPE_INTEGER) {
```

Comment:
Already pushed code to refactor that function @aelroby please pull from my repo.